src/app/features/notification/notification.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from "@angular/core";
+import "firebase/compat/messaging";
+import { getMessaging, getToken, onMessage } from "firebase/messaging";
+import { Logging } from "app/core/logging/logging.service";
+import { HttpClient } from "@angular/common/http";
+import { environment } from "../../../environments/environment";
+import firebase from "firebase/compat/app";
+
+/**
+ * Handles the interaction with Cloud Messaging.
+ * It manages the retrieval of Cloud Messaging Notification token, listens for incoming messages, and sends notifications
+ * to users. The service also provides methods to create cloud messaging payloads and communicate with the
+ * cloud messaging HTTP API for sending notifications.
+ */
+@Injectable({
+  providedIn: "root",
+})
+export class NotificationService {
+  private messaging: any = null;
+  readonly FIREBASE_CLOUD_MESSAGING_URL = `https://fcm.googleapis.com/v1/projects/${environment.firebaseConfig.projectId}/messages:send`;
+
+  constructor(private httpClient: HttpClient) {
+    firebase.initializeApp(environment.firebaseConfig);
```

Comment:
Is this different from what you do in NotificationModule with `initializeApp(environment.firebaseConfig);`?

Reply:
and how is it different/connected with the initialization in src/firebase-messaging-sw.js?

Reply:
> Is this different from what you do in NotificationModule with `initializeApp(environment.firebaseConfig);`?

Thank you for pointing this out. Both achieve the same purpose, so I have removed `initializeApp(environment.firebaseConfig);` from the module since the notification service file handles this initialization.

 > and how is it different/connected with the initialization in src/firebase-messaging-sw.js?
 
As per my understanding, the `src/firebase-messaging-sw.js` is primarily used to handle background notification messages. To enable this functionality, Firebase needs to be initialized specifically for background messages. When I tried removing the initialization from the service worker, I encountered an error: `FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app-compat/no-app)`.